folium/colormap.py

```diff
@@ -0,0 +1,594 @@
+                    '#F768A1', '#DD3497', '#AE017E', '#7A0177'],
+           'YlGn': ['#FFFFCC', '#D9F0A3', '#ADDD8E',
+                    '#78C679', '#41AB5D', '#238443', '#005A32'],
+           'YlGnBu': ['#FFFFCC', '#C7E9B4', '#7FCDBB',
+                      '#41B6C4', '#1D91C0', '#225EA8', '#0C2C84'],
+           'YlOrBr': ['#FFFFD4', '#FEE391', '#FEC44F',
+                      '#FE9929', '#EC7014', '#CC4C02', '#8C2D04'],
+           'YlOrRd': ['#FFFFB2', '#FED976', '#FEB24C',
+                      '#FD8D3C', '#FC4E2A', '#E31A1C', '#B10026'],
+           'BrBg': ['#8c510a', '#d8b365', '#f6e8c3',
+                    '#c7eae5', '#5ab4ac', '#01665e'],
+           'PiYG': ['#c51b7d', '#e9a3c9', '#fde0ef',
+                    '#e6f5d0', '#a1d76a', '#4d9221'],
+           'PRGn': ['#762a83', '#af8dc3', '#e7d4e8',
+                    '#d9f0d3', '#7fbf7b', '#1b7837'],
+           'PuOr': ['#b35806', '#f1a340', '#fee0b6',
+                    '#d8daeb', '#998ec3', '#542788'],
+           'RdBu': ['#b2182b', '#ef8a62', '#fddbc7',
+                    '#d1e5f0', '#67a9cf', '#2166ac'],
+           'RdGy': ['#b2182b', '#ef8a62', '#fddbc7',
+                    '#e0e0e0', '#999999', '#4d4d4d'],
+           'RdYlBu': ['#d73027', '#fc8d59', '#fee090',
+                      '#e0f3f8', '#91bfdb', '#4575b4'],
+           'RdYlGn': ['#d73027', '#fc8d59', '#fee08b',
+                      '#d9ef8b', '#91cf60', '#1a9850'],
+           'Spectral': ['#d53e4f', '#fc8d59', '#fee08b',
+                        '#e6f598', '#99d594', '#3288bd'],
+           'Accent': ['#7fc97f', '#beaed4', '#fdc086',
+                      '#ffff99', '#386cb0', '#f0027f'],
+           'Dark2': ['#1b9e77', '#d95f02', '#7570b3',
+                     '#e7298a', '#66a61e', '#e6ab02'],
+           'Paired': ['#a6cee3', '#1f78b4', '#b2df8a',
+                      '#33a02c', '#fb9a99', '#e31a1c'],
+           'Pastel1': ['#fbb4ae', '#b3cde3', '#ccebc5',
+                       '#decbe4', '#fed9a6', '#ffffcc'],
+           'Pastel2': ['#b3e2cd', '#fdcdac', '#cbd5e8',
+                       '#f4cae4', '#e6f5c9', '#fff2ae'],
+           'Set1': ['#e41a1c', '#377eb8', '#4daf4a',
+                    '#984ea3', '#ff7f00', '#ffff33'],
+           'Set2': ['#66c2a5', '#fc8d62', '#8da0cb',
+                    '#e78ac3', '#a6d854', '#ffd92f'],
+           'Set3': ['#8dd3c7', '#ffffb3', '#bebada',
+                    '#fb8072', '#80b1d3', '#fdb462'],
+          }
+
+def _parse_color(x):
+    """
+    """
+    if isinstance(x, tuple) or isinstance(x, list):
+        color_tuple = tuple(x)[:4]
+    elif isinstance(x, text_type) or isinstance(x, binary_type):
+        if x[0]=="#" and len(x)==7:
+            # Color of the for #RRGGBB
+            color_tuple = (int(x[1:3],16), int(x[3:5],16), int(x[5:7],16))
+        else:
+            color_code = _cnames.get(x.lower(),None)
+            if color_code is None:
+                raise ValueError('Unknown color.')
```

Comment:
It always help people to debug if we print the `{!r}` of the offending object.  How about:
`'Unknown color {!r}.'.format(x)`